dm/config/subtask.go

```diff
@@ -268,8 +269,11 @@ func (c *SubTaskConfig) Adjust(verifyDecryptPassword bool) error {
 		c.ShardMode = ShardPessimistic // use the pessimistic mode as default for back compatible.
 	}
 
-	if c.OnlineDDLScheme != "" && c.OnlineDDLScheme != PT && c.OnlineDDLScheme != GHOST {
+	if !c.OnlineDDL && c.OnlineDDLScheme != "" && c.OnlineDDLScheme != PT && c.OnlineDDLScheme != GHOST {
 		return terror.ErrConfigOnlineSchemeNotSupport.Generate(c.OnlineDDLScheme)
+	} else if c.OnlineDDLScheme == PT || c.OnlineDDLScheme == GHOST {
```

Comment:
subtask is generated from task, so we may omit this check and only check at task level

Reply:
In that case, can the above two lines be omitted?
https://github.com/pingcap/dm/pull/1904/files/eb17460e69c7d7428d1067e01c64d71a058871a9#diff-582afcdf0d8ec3525c230111a96de59d175e67d9435b23713384447a07ee4547R272-R273

Reply:
If we delete this check, the test will fail. https://github.com/pingcap/dm/pull/1904#discussion_r676261884

Reply:
> If we delete this check, the test will fail. [#1904 (comment)](https://github.com/pingcap/dm/pull/1904#discussion_r676261884)

I am sorry I misunderstood. Deleting this check does not lead the test to fail.